src/command.ts

```diff
@@ -29,12 +39,15 @@ abstract class Command {
 	parent: BotCommand = null;
 	altNames: string[] = [];
 	help: { msg?: string; usage?: string; } = {}
+	slashCommand?: boolean = false;
+	slashOptions?: SlashCommandOption[] = [];
```

Comment:
Slash commands should be their own class that extends the Command class with these fields as required.